File: LC-Python-Solution/24. Swap Nodes in Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)
        dummy.next = head
        prev = dummy

        while(head and head.next):
            FirstNode = head
            SecondNode = head.next

            #SWAPING TWO ADJACENT NODES

            FirstNode.next = SecondNode.next
            SecondNode.next = FirstNode
            prev.next = SecondNode

            #Moving to other Two Node

            prev = FirstNode
            head = FirstNode.next
        
        return dummy.next

File: LC_JS-SOLUTIONS/24. Swap Nodes in Pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    const dummy =new ListNode(0);
    dummy.next = head;
    let prev = dummy;

    while(head && head.next){
        let FirstNode = head
        let SecondNode = head.next

        FirstNode.next =SecondNode.next
        SecondNode.next = FirstNode
        prev.next = SecondNode

        prev = FirstNode
        head =  FirstNode.next
    }
    return dummy.next



};